### ISSUE TITLE
`pnpm` Does Not Recognize `.npmrc` in Sub-Project When Running Scripts from Parent Project

### ISSUE THREAD

**Issue by Innei** [pnpm/pnpm#7550]
### Verify latest release

- [x] I verified that the issue exists in the latest pnpm release

### pnpm version

8.14.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/Innei/pnpm-npmrc-location-reproduction

### Reproduction steps

1. clone reproduction
2. run pnpm i
3. see the console output of `pnpm config list` result

### Describe the Bug

1. A my-project directory with its own `.npmrc` file set to `node-linker=hoisted`.
2. Inside my-project, you have a package.json file that triggers a postinstall script: "postinstall": "sh ./scripts/install.sh".
3. The install.sh script changes directory to pkg1 and then runs `pnpm config list`.
The pkg1 directory has its own .npmrc file with `node-linker=isolated`.
4. However, when you run `pnpm i` in my-project, the output shows that the node-linker is set to hoisted instead of isolated as defined in pkg1/.npmrc.

This behavior indicates that pnpm is not recognizing the `.npmrc` configuration of the pkg1 directory when the script executed by my-project changes its working directory to pkg1. It retains the configuration of the original project (my-project) from which the script was executed.

### Expected Behavior

When running pnpm i in my-project, the postinstall script should recognize and use the .npmrc configuration of the pkg1 directory, setting node-linker to isolated.

### Which Node.js version are you using?

v20

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by SoonIter** [comment]
Many configurations of pnpm, including `.npmrc` `pnpm.overrides` `.pnpmfile.cjs`, are centralized, and the whole repo has only one I think. So do the installation.

**Comment by zkochan** [comment]
But if you add some dependencies to pkg1 you'll see that an isolated node_modules will be created, as configured in `.npmrc`.

**Comment by Innei** [comment]
> Many configurations of pnpm, including `.npmrc` `pnpm.overrides` `.pnpmfile.cjs`, are centralized, and the whole repo has only one. So do the installation and `pnpm-lock.yaml`

No, this is not a monorepo. You can understand it as two separate projects. The node_module and npmrc should exist independently. My current issue is that the behavior of executing install.sh through script and directly executing the install.sh(copy commands into shell to exec) command is different.

**Comment by Innei** [comment]
> But if you add some dependencies to pkg1 you'll see that an isolated node_modules will be created, as configured in `.npmrc`.

No, I add a dependency like `express` in `pkg1`, and then execute `pnpm i` in the root directory. The management of node_modules in pkg1 is not isolated.
![CleanShot 2024-01-23 at 11  03 10@2x](https://github.com/pnpm/pnpm/assets/41265413/8594b662-0ac8-4831-b5c9-602b7030be2f)

**Comment by zkochan** [comment]
I think you mean when you add the install in pkg1 to the postinstall of the root project. Because if you just run install in the root, pnpm won't install dependencies in pkg1.

**Comment by Innei** [comment]
> I think you mean when you add the install in pkg1 to the postinstall of the root project. Because if you just run install in the root, pnpm won't install dependencies in pkg1.

Yes. But this installation behavior is not based on `.npmrc` under pkg1.

**Comment by zkochan** [comment]
This targets a part of pnpm that no longer exists, so closing.

pnpm settings are read from pnpm-workspace.yaml, one file for the whole workspace, so a project-level .npmrc no longer carries them.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
